flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/JobAutoScalerImpl.java

```diff
@@ -79,6 +85,8 @@ public boolean scale(FlinkResourceContext<? extends AbstractFlinkResource<?, ?>>
         var conf = ctx.getObserveConfig();
         var resource = ctx.getResource();
         var resourceId = ResourceID.fromResource(resource);
+        var flinkMetrics = getOrInitAutoscalerFlinkMetrics(ctx, resourceId);
```

Comment:
we could now use the `flinkMetrics` in the catch block instead of reinitializing it